forgerock-core/src/main/java/org/forgerock/android/auth/InterceptorProvider.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ *
+ *  * Copyright (c) 2022 ForgeRock. All rights reserved.
+ *  *
+ *  * This software may be modified and distributed under the terms
+ *  * of the MIT license. See the LICENSE file for details.
+ *
+ *
+ */
+
+package org.forgerock.android.auth
+
+import okhttp3.logging.HttpLoggingInterceptor
+
+internal class InterceptorProvider @JvmOverloads constructor(private val customLogger: FRLogger? = null,
+                                                             private val isDebugEnabled: Boolean = false) {
+    fun getInterceptor(): HttpLoggingInterceptor? {
+        val httpLogger = object: HttpLoggingInterceptor.Logger {
+            override fun log(message: String) {
+                customLogger?.network("OKHttpClient..", message, "")
+            }
+        }
+        if(customLogger != null) {
```

Comment:
Is there a way that we don't inject the interceptor for customLogger, With the HttpLoggingInterceptor injected to OKHttpClient, I think there are quite a lot of string manipulation and logic, would like to minimize the overhead when not necessary. May be an interface IsNetworkEnabled in FRLogger interface?

Reply:
@witrisna we can brainstorm this may be tomorrow, i would like to understand your thought process